### PR TITLE
Assignment3

### PR DESCRIPTION
**Pull request by Aedyn01**
Submission of Assignment 3, team member is @mpennock5

### CODE REVIEW

File: CanHazFunny/CanHazFunny.Tests/JokeServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class JokeServiceTests
+{
+    [TestMethod]
+    public void JokeService_GetJokeReturnsAJoke_True()
+    {
+        Mock<JokeService> mockJoke = new();
+      
+        mockJoke.SetupSequence(joke => joke.GetJoke()).Returns("this is a joke, ha, ha, ha");
+
+        Assert.AreEqual<string?>("this is a joke, ha, ha, ha", mockJoke.Object.GetJoke());
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny.Tests/JokeServiceTests.cs:21]: Something to keep in mind, this isn't actually testing your GetJoke method, it's testing that your mock returns a string. Mocks are useful when using them to test something that relies on the mock to better isolate code.

File: CanHazFunny/CanHazFunny/JokeService.cs [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-            string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
-            return joke;
-        }
+    public virtual string GetJoke()
+    {
+        
+        string joke = HttpClient.GetStringAsync("https://geek-jokes.sameerkumar.website/api").Result;
+                       
+        return joke;
     }
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/JokeService.cs:15]: There is odd spacing going on in this class.

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public Jester(IJokeOutput jokeOutput, IJokeService jokeService)
+    {
+        JokeOutput = jokeOutput ?? throw new ArgumentNullException(nameof(jokeOutput));
+        JokeService = jokeService ?? throw new ArgumentNullException(nameof(jokeService));
+    }
+
+    public IJokeOutput JokeOutput { get; }
+    public IJokeService JokeService { get; }
+
+    public string? Joke { get; set; }
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:16]: I would suggest keeping this as a locally scoped variable to the `TellJoke` method.

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    public virtual void TellJoke()
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:18]: Does this need to be virtual?

**Comment by mmwoodfo** [comment]
I can't find a PR review, did you do one?

**Comment by Aedyn01** [comment]
I thought I tagged mpennock5 as my teammate for this assignment. We both
worked on this so he should receive credit for it

**Pull request closed without merging**